src/adapter/src/coord.rs

```diff
@@ -2594,12 +2594,18 @@ impl Coordinator {
 
             write_frontier = self.least_valid_write(&storage_dependencies);
             if let Some(ts) = write_frontier.as_option() {
+                // Make sure all storage dependencies are readable at the chosen
+                // as_of by stepping back by one from the least write frontier
+                // at the inputs. This makes sure we get a readable timestamp
+                // regardless of the compaction_window.
+                let ts = ts.step_back().unwrap_or_else(Timestamp::minimum);
```

Comment:
I'm surprised that this is needed! Can `compaction_window` ever be zero? AFAIK, various things start to go wrong if we ever select a compaction window of zero, [but we just agree to not allow that](https://github.com/MaterializeInc/materialize/issues/13221#issuecomment-1219851030). That comment is pretty old, so maybe that has changed more recently?